icu-date.lua

```diff
@@ -51,7 +51,7 @@ local function string_to_uchar(str)
 end
 
 local function uchar_to_string(uchar)
-  local length = call_fn("u_strlen", uchar) + 1
+  local length = call_fn("u_strlen", uchar) + 2
```

Comment:
Looks a bit strange. Could you describe why did you choose 2? Why not 4?

Reply:
u_strlen():
> Returns
The number of UChars in chars, minus the terminator.

I guess an issue here. You need to get length in bytes, not in chars. Try to use just simple strlen instead of u_strlen.

Reply:
no excuses, just try fix it.